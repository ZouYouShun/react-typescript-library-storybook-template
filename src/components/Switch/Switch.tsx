import React, { forwardRef, ReactNode } from 'react';

type UISwitchProps = {
  /** children render in the component */
  children?: ReactNode;
};

const UISwitch = forwardRef<any, UISwitchProps>(
  ({ children, ...rest }, ref) => {
    return (
      <div {...rest} ref={ref}>
        {children}
      </div>
    );
  },
);

UISwitch.defaultProps = {};

UISwitch.displayName = 'UISwitch';

export { UISwitch, UISwitchProps };
