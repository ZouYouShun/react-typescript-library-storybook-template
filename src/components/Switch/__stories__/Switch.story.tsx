import { Meta, Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';

import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
} from '../../../storybook';
import { UISwitch } from '../Switch';

export default {
  title: 'Components/Switch',
  component: UISwitch,
  argTypes: {
    ...sortInDocTable<keyof SwitchProps>([]),
    ...notControlInDocTable<keyof SwitchProps>([]),
    ...notShowInDocTable<keyof SwitchProps>([]),
  },
} as Meta;

type SwitchProps = ComponentProps<typeof UISwitch>;

export const Switch: Story<SwitchProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return <UISwitch {...args}>{children}</UISwitch>;
};

Switch.args = {
  children: 'Title',
};

Switch.argTypes = {
  ...notControlInDocTable<keyof SwitchProps>([]),
};

Switch.parameters = {
  tags: [
    {
      name: 'Spec',
      href: '',
    },
    {
      name: 'Mui',
      href: '',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};
