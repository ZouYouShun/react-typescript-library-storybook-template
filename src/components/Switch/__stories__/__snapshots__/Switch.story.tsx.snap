// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/Switch Switch 1`] = `
<div>
  Title
</div>
`;
